examples/threathunter/threat_intelligence/stix_taxii.py

```diff
@@ -0,0 +1,283 @@
+            client = create_client(conf.site,
+                                   use_https=conf.use_https,
+                                   discovery_path=conf.discovery_path)
+            client.set_auth(username=conf.username,
+                            password=conf.password,
+                            verify_ssl=conf.ssl_verify,
+                            ca_cert=conf.ca_cert,
+                            cert_file=conf.cert_file,
+                            key_file=conf.key_file)
+
+            proxy_dict = dict()
+            if conf.http_proxy_url:
+                proxy_dict['http'] = conf.http_proxy_url
+            if conf.https_proxy_url:
+                proxy_dict['https'] = conf.https_proxy_url
+            if proxy_dict:
+                client.set_proxies(proxy_dict)
+
+            self.client = client
+
+        except handled_exceptions as e:
+            logging.error(f"Error creating client: {e}")
+
+
+
+    def create_uri(self, config_path):
+        uri = None
+        if self.config.site and config_path:
+            if self.config.use_https:
+                uri = 'https://'
+            else:
+                uri = 'http://'
+            uri = uri + self.config.site + config_path
+        return uri
+
+
+
+    def query_collections(self):
+        collections = []
+        try:
+            uri = self.create_uri(self.config.collection_management_path)
+            collections = self.client.get_collections(
+                uri=uri)  # autodetect if uri=None
+            for collection in collections:
+                logging.debug(f"Collection: {collection.name}, {collection.type}")
+        except handled_exceptions as e:
+            logging.warning(f"Problem fetching collections from taxii server: {e}")
+        return collections
+
+    def poll_server(self, collection, feed_helper):
+        content_blocks = []
+        uri = self.create_uri(self.config.poll_path)
+        try:
+            logging.info(f"Polling Collection: {collection.name}")
+            content_blocks = self.client.poll(
+                uri=uri,
+                collection_name=collection.name,
+                begin_date=feed_helper.start_date,
+                end_date=feed_helper.end_date,
+                content_bindings=BINDING_CHOICES)
+        except handled_exceptions as e:
+            logging.warning(f"problem polling taxii server: {e}")
+        return content_blocks
+
+    def parse_collection_content(self, content_blocks):
+        for block in content_blocks:
+            yield from parse_stix(block.content, self.config.default_score)
+
+    def import_collection(self, collection):
+        num_times_empty_content_blocks = 0
+        advance = True
+        reports_limit = self.config.reports_limit
+        logging.info(f"reports limit: {reports_limit}")
+        feed_helper = FeedHelper(self.config.start_date,
+                                 self.config.minutes_to_advance)
+
+        while feed_helper.advance():
```

Comment:
I don't really understand the intention behind this while loop condition. Are we trying to poll for the amount of time that we are advancing? Or are we trying to advance every X minutes?

Reply:
This while loop condition (feed_helper.advance) is used to determine if we've hit the end of our TAXII Collection. The config parameters `start_date` and `minutes_to_advance` tell this Feed Helper where to start polling in the collection, and then will poll in chunks of `minutes_to_advance` minutes until we hit the most current `content_block`.

Reply:
I've commented stix_taxii.py to be more clear on this